### PR TITLE
UD/TEST: Set minimal queue length when running with valgrind.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by shamisp** [comment]
why do we need this ?

**Comment by yosefe** [comment]
to make jenkins complete faster

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1191/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/614/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
autobot build

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1193/ for details.

**Comment by shamisp** [comment]
okay…our testing machines are down due to storm ...

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/616/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1195/ for details.

**Comment by yosefe** [comment]
@shamisp @miked-mellanox pls review
the test failure is unrelated

File: src/ucs/gtest/main.cc [hunk 1]
```diff
@@ -37,5 +50,13 @@ int main(int argc, char **argv) {
     }
     UCS_TEST_MESSAGE << "Using random seed of " << ucs_gtest_random_seed;
     srand(ucs_gtest_random_seed);
+    if (RUNNING_ON_VALGRIND) {
+        modify_config_for_valgrind("IB_RX_QUEUE_LEN", "512");
+        modify_config_for_valgrind("IB_RX_BUFS_GROW", "512");
+        modify_config_for_valgrind("MM_RX_BUFS_GROW", "32");
```
Comment (shamisp) [inline src/ucs/gtest/main.cc]: For consistency I would add one for ugni

File: src/uct/tl/tl_base.h [hunk 2]
```diff
@@ -200,7 +200,6 @@ struct uct_iface_config {
         uct_alloc_method_t  *methods;
         unsigned            count;
     } alloc_methods;
-    UCS_CONFIG_ARRAY_FIELD(ucs_range_spec_t, signals) lid_path_bits;
```
Comment (shamisp) [inline src/uct/tl/tl_base.h]: I think, this is unrelated change ? Should it go as a separate PR ?

File: src/ucs/gtest/main.cc (hunk 1 above)
Comment (yosefe) [inline src/ucs/gtest/main.cc]: ok
what is the parameter name for UGNI?

File: src/uct/tl/tl_base.h (hunk 2 above)
Comment (yosefe) [inline src/uct/tl/tl_base.h]: it's removing a configuration that was put there by mistake

Comment (shamisp) [inline src/uct/tl/tl_base.h]: I realized so. For now it is ok, but in feature I would suggest a separate PR, so we don't confuse things.

File: src/ucs/gtest/main.cc (hunk 1 above)
Comment (shamisp) [inline src/ucs/gtest/main.cc]: UGNI_BUFS_GROW ?

Comment (yosefe) [inline src/ucs/gtest/main.cc]: not sure. i believe you have several memory pools there
and what about queue lengths?
can you run ucx_info -c on a machine with ugni to collect all variables you want to change here?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1196/ for details.

**Comment by miked-mellanox** [comment]
:+1:
